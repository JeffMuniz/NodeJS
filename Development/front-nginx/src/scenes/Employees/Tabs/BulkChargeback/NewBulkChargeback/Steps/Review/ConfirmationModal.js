import React from "react";
import { func, shape } from "prop-types";
import { get } from "lodash";
import { connect } from "react-redux";

import { NewModal } from "@base";
import { toMoneyMask, toMoney } from "@utils";
import { ChargeBackTOS } from "@common";

import * as ModalActions from "src/redux/modules/modal/actions/modal";
import {
  ContentWrapper,
  TotalValue,
  StyledParagraph,
  Bold,
} from "./ConfirmationModal.styled";

const getTotalValue = employees =>
  employees.reduce(
    (acc, curr) =>
      acc + toMoney(get(curr, "va", 0)) + toMoney(get(curr, "vr", 0)),
    0,
  );

const getmacefitsTypes = employees => {
  const vaString = "mac Alimentação";
  const vrString = "mac Refeição";
  const value = "R$ 0,00".toString();
  const hasVa = employees.some(el => el.va !== value && el.va !== undefined);
  const hasVr = employees.some(el => el.vr !== value && el.vr !== undefined);

  return `${hasVr ? vrString : ""}${hasVa && hasVr ? " e " : ""}${
    hasVa ? vaString : ""
  }`;
};

export const ConfirmationModal = ({
  openModal,
  closeModal,
  values,
  companySelected,
  goBackDetails,
}) => {
  const employees = get(values, "cpfs", []);
  const total = getTotalValue(employees);
  const [{ reason }] = employees;
  const totalOfEmployees = employees.length;
  const macefitTypeString = getmacefitsTypes(employees);
  const onClick = () =>
    openModal({
      children: (
        <ChargeBackTOS
          employees={values}
          company={companySelected}
          redirect={goBackDetails}
        />
      ),
    });

  return (
    <NewModal
      title="Confirmação de estorno"
      subtitle="Verifique os dados abaixo para realizar o estorno."
      id="modal-confirmation"
      width="706px"
      height="unset"
      okButtonProps={{
        value: "Confirmar",
        id: "btn-modal-confirmation-ok",
        disabled: total === 0,
        onClick,
      }}
      cancelButtonProps={{
        value: "Cancelar",
        onClick: closeModal,
        id: "btn-modal-confirmation-cancel",
      }}
    >
      <ContentWrapper id="content-modal-confirmation">
        <StyledParagraph margin="5px">Valor total solicitado:</StyledParagraph>
        <TotalValue id="field-total-value">{toMoneyMask(total)}</TotalValue>
        <StyledParagraph>
          Motivo do estorno: <Bold id="field-reason">{reason}</Bold>
        </StyledParagraph>
        <StyledParagraph>
          Quantidade de funcionários:{" "}
          <Bold id="field-total-employees">{totalOfEmployees}</Bold>
        </StyledParagraph>
        <StyledParagraph>
          Produtos: <Bold id="field-product-type">{macefitTypeString}</Bold>
        </StyledParagraph>
      </ContentWrapper>
    </NewModal>
  );
};

ConfirmationModal.propTypes = {
  openModal: func.isRequired,
  closeModal: func.isRequired,
  values: shape({}).isRequired,
  companySelected: shape({}).isRequired,
  goBackDetails: func.isRequired,
};

const mapStateToProps = ({ selectedCompanyTree }) => ({
  companySelected: selectedCompanyTree.selectedCompany,
});

const mapDispatchToProps = {
  openModal: ModalActions.OpenModal,
  closeModal: ModalActions.CloseModal,
};

export default connect(mapStateToProps, mapDispatchToProps)(ConfirmationModal);
