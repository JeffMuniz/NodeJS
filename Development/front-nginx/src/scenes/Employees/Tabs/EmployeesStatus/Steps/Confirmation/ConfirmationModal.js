import React from "react";
import { arrayOf, func, shape } from "prop-types";

import { NewModal } from "@base";

import {
  Bold,
  ContentWrapper,
  StyledParagraph,
} from "./ConfirmationModal.styled";

export const ConfirmationModal = ({
  closeModal,
  values,
  handleSubmitModal,
}) => {
  const [{ reasonDescription, status }] = values;
  const employeesTotal = values.length;
  const textStatus = status ? "inativação" : "ativação";

  return (
    <NewModal
      title={`Confirmação da ${textStatus}`}
      subtitle={`Verifique os dados abaixo para realizar a ${textStatus}`}
      id="modal-confirmation-change-status"
      width="600px"
      height="unset"
      okButtonProps={{
        value: "Confirmar",
        id: "btn-modal-confirmation-ok",
        disabled: employeesTotal === 0,
        onClick: () => handleSubmitModal(values),
      }}
      cancelButtonProps={{
        value: "Cancelar",
        onClick: closeModal,
        id: "btn-modal-confirmation-cancel",
      }}
    >
      <ContentWrapper id="content-modal-confirmation">
        <StyledParagraph>
          Motivo da {textStatus}:{" "}
          <Bold id="field-reason">{reasonDescription}</Bold>
        </StyledParagraph>
        <StyledParagraph margin="30px">
          Quantidade de funcionários:{" "}
          <Bold id="field-total-employees">{employeesTotal}</Bold>
        </StyledParagraph>
        {status ? (
          <StyledParagraph>
            A partir do momento que você confirmar a inativação, <br />
            esses funcionários deixarão de receber a(s) carga(s) em <br />
            seu(s) cartão(s), até ele ser ativado novamente
          </StyledParagraph>
        ) : (
          <StyledParagraph>
            A partir do momento que você confirmar a ativação, esses <br />
            funcionários voltarão a receber a(s) carga(s) em seu(s) <br />
            cartão(s) novamente
          </StyledParagraph>
        )}
      </ContentWrapper>
    </NewModal>
  );
};

ConfirmationModal.propTypes = {
  closeModal: func.isRequired,
  values: arrayOf(shape({})).isRequired,
  handleSubmitModal: func.isRequired,
};

export default ConfirmationModal;
