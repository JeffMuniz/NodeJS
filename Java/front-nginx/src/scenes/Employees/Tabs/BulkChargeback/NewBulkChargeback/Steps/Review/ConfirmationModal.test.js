import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";
import { ConfirmationModal } from "./ConfirmationModal";
import cpfs from "./__mock__/data";

const defaultProps = {
  openModal: jest.fn(),
  closeModal: jest.fn(),
  values: { cpfs },
};

describe("NewBulkChargeback - ConfirmationModal", () => {
  it("Should be ConfirmationModal", () => {
    const renderer = new ShallowRenderer();

    // when
    renderer.render(<ConfirmationModal {...defaultProps} />);
    const result = renderer.getRenderOutput();

    // then
    expect(result).toMatchSnapshot();
  });
});
