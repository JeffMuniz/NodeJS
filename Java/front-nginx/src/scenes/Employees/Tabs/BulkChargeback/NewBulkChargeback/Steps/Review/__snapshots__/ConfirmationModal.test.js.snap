// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewBulkChargeback - ConfirmationModal Should be ConfirmationModal 1`] = `
<Modal
  cancelButtonProps={
    Object {
      "id": "btn-modal-confirmation-cancel",
      "onClick": [MockFunction],
      "value": "Cancelar",
    }
  }
  contentHasScroll={false}
  height="unset"
  id="modal-confirmation"
  isView={false}
  maxHeight={null}
  okButtonProps={
    Object {
      "disabled": false,
      "id": "btn-modal-confirmation-ok",
      "onClick": [Function],
      "value": "Confirmar",
    }
  }
  onContentBottomReach={null}
  padding={null}
  subtitle="Verifique os dados abaixo para realizar o estorno."
  title="Confirmação de estorno"
  viewButtonProps={Object {}}
  width="706px"
>
  <styled.div
    id="content-modal-confirmation"
  >
    <styled.p
      margin="5px"
    >
      Valor total solicitado:
    </styled.p>
    <styled.p
      id="field-total-value"
    >
      R$ 2,00
    </styled.p>
    <styled.p>
      Motivo do estorno: 
      <styled.span
        id="field-reason"
      >
        demissao
      </styled.span>
    </styled.p>
    <styled.p>
      Quantidade de funcionários:
       
      <styled.span
        id="field-total-employees"
      >
        4
      </styled.span>
    </styled.p>
    <styled.p>
      Produtos: 
      <styled.span
        id="field-product-type"
      >
        mac Refeição e mac Alimentação
      </styled.span>
    </styled.p>
  </styled.div>
</Modal>
`;
